### PR TITLE
FOGL-1801: FogLAMP Benchmarking activities

### PR DESCRIPTION
**Pull request by AmandeepArora**
Let me know if I should retain the omf.h and omf.cpp changes or not.

### CODE REVIEW

File: C/plugins/common/omf.cpp [hunk 1]
```diff
@@ -319,8 +323,13 @@ uint32_t OMF::sendToServer(const vector<Reading *>& readings,
 	// Then get HTTPS POST ret code and return 0 to client on error
 	try
 	{
+		std::chrono::high_resolution_clock::time_point t1 = std::chrono::high_resolution_clock::now();
```
Comment (MarkRiddoch) [inline C/plugins/common/omf.cpp]: I wonder how expensive it is to track the time?

File: C/services/south/south.cpp [hunk 2]
```diff
@@ -202,8 +202,11 @@ void SouthService::start(string& coreAddress, unsigned short corePort)
 
 		try {
 			m_pollInterval = 500;
+			m_readingsPerSec = 1;
 			if (m_config.itemExists("pollInterval"))
```
Comment (MarkRiddoch) [inline C/services/south/south.cpp]: We can probably ditch the pollInterval stuff completely and just have the readingsPerSec for today

File: C/plugins/common/omf.cpp (hunk 1 above)
Comment (AmandeepArora) [reply C/plugins/common/omf.cpp]: I don't think much, but I can remove it if you wish. Or I can find out the overhead.

Comment (AmandeepArora) [reply C/plugins/common/omf.cpp]: Actually, when we have sent out 5000 readings to Pi server, it takes around 700-1000 msec to get POST request serviced, so in that sense overhead of time calculation is insignificant.

File: C/services/south/south.cpp (hunk 2 above)
Comment (AmandeepArora) [reply C/services/south/south.cpp]: Done.

File: C/plugins/common/omf.cpp (hunk 1 above)
Comment (AmandeepArora) [reply C/plugins/common/omf.cpp]: Tried it on my ubuntu VM, I got "1000 times duration calc took 213 usecs" and "100000 times duration calc took 17590 usecs". So it's around 175-213 nano-secs.

        std::chrono::high_resolution_clock::time_point t1 = std::chrono::high_resolution_clock::now();
	std::chrono::high_resolution_clock::time_point t2;
	int i;
	for (i=0; i<1000; i++)
		t2 = std::chrono::high_resolution_clock::now();
 	auto usecs = std::chrono::duration_cast<std::chrono::microseconds>( t2 - t1 ).count();
	logger->info("%d times duration calc took %lld usecs", i, usecs);

Comment (MarkRiddoch) [reply C/plugins/common/omf.cpp]: OK, fine thanks.

**Pull request merged**